aten/src/ATen/native/UnaryOps.cpp

```diff
@@ -145,14 +145,28 @@ TORCH_IMPL_FUNC(func_out) (const Tensor& self, const Tensor& result) {  \
   func_stub(device_type(), *this);                                      \
 }
 
+// This macro is as optional as the one above. torch.(ceil|floor|round|trunc) are no-ops for integers
+// See gh-70918
+#define CREATE_UNARY_TORCH_IMPL_INTEGER_NO_OP_FUNC(func_out, func_stub)                                \
+TORCH_IMPL_FUNC(func_out) (const Tensor& self, const Tensor& result) {  \
+  if (c10::isIntegralType(self.scalar_type(), /*includeBool=*/false)) {                                      \
```

Comment:
Why not include bools here?

Reply:
Converting bools but not integer types seems very strange to me -- but good point. Why don't we also return self for bools but add a comment that it's inconsistent with NumPy's behavior

Reply:
Another data-point to consider,

Array-API mentions supporting only Numeric Dtype (without boolean)

round: https://data-apis.org/array-api/latest/API_specification/generated/signatures.elementwise_functions.round.html?highlight=round

ceil: https://data-apis.org/array-api/latest/API_specification/generated/signatures.elementwise_functions.ceil.html?highlight=ceil

Numeric Datatypes: https://data-apis.org/array-api/latest/API_specification/data_types.html#numeric-data-types

Reply:
> Another data-point to consider,
> 
> Array-API mentions supporting only Numeric Dtype (without boolean)
> 
> round: https://data-apis.org/array-api/latest/API_specification/generated/signatures.elementwise_functions.round.html?highlight=round
> 
> ceil: https://data-apis.org/array-api/latest/API_specification/generated/signatures.elementwise_functions.ceil.html?highlight=ceil
> 
> Numeric Datatypes: https://data-apis.org/array-api/latest/API_specification/data_types.html#numeric-data-types

Rather than returning anything then perhaps the current behavior is alright, but should have a better error? (currently gives a non-informative error on not being implemented for `bool`)

Reply:
What's non-informative about "not implemented for Bool" error? It provides all the necessary information

Reply:
> What's non-informative about "not implemented for Bool" error? It provides all the necessary information

Just that it doesn't explain why (in this case, because the array-api doesn't cover non-numeric types for these functions).

Otherwise, it might be assumed that `not implemented for Bool` means there might be scope to add an implementation (e.g. returning `self` for `bool`).

Reply:
Every error message shouldn't give brief history and discussions of the error in question, enough to say what went wrong.

Reply:
> Every error message shouldn't give brief history and discussions of the error in question, enough to say what went wrong.

That makes sense too.


So for the `bool` types we have some implementation decisions:

- Leave things as they are (throw an error for `bool`, in keeping with the `array-api`)
- Return `self` for `bool` (documenting that this is a break from the `array-api` and NumPy)
- Implement the NumPy approach of converting to  `float16` and then returning `self`

Reply:
Throwing an error is the safest and most future-proof of these options.